### ISSUE TITLE
Round floats in annotation xmls and warn

### ISSUE THREAD

**Issue by bw4sz** [weecology/DeepForest#104]
from https://github.com/weecology/DeepForest/issues/102
@aasem commented

> I debugged through the utilities.py and changed int typecast of annotations to float since, in my case, exported JSON or xml were in floating point. But when I changed the typecast, there were errors so I changed utilities.py again and rounded all my floats and typecasted back to int. 

Can you provide an annotations file that you were debugging. I can write some tests and work on the float versus integer workflow. I'll need to think more about it, but floats seem not possible in my current conception since we are talking about the image coordinate system. The anchor boxes cannot be placed between pixels, so they must span  the entire pixel. DeepForest could
 
1. Warn that the incoming XML is float and round automatically.
2. Raise an error

and the docs should be updated stating that float annotations aren't possible in the image coordinate system. I made a small gist for converting from a projected coordinate system, where of course floats are fine, but I mostly leave that to user since its impossible to anticipate all complexities.

https://gist.github.com/bw4sz/e2fff9c9df0ae26bd2bfa8953ec4a24c

**Comment by aasem** [comment]
> Can you provide an annotations file that you were debugging. >

I have attached a portion of my xml file: [tile.txt](https://github.com/weecology/DeepForest/files/4466159/tile.txt)

> I can write some tests and work on the float versus integer workflow. I'll need to think more about it, but floats seem not possible in my current conception since we are talking about the image coordinate system. The anchor boxes cannot be placed between pixels, so they must span the entire pixel.>

I think a warning would do. If it's useful, I might add that I used [labelme](https://github.com/wkentaro/labelme) to generate VOC-format Dataset. It gave floating points. You are right that all complexities cannot be anticipated but it floating point pixel precision do come up in computations nevertheless. For instance, computing bounding box parameter-set from polygon centers of an already labeled data from Global Mapper or any other GIS software.
